### PR TITLE
feat: add event logging api and more tracing

### PR DESCRIPTION
**Pull request by lalo**
https://github.com/microsoft/autogen/pull/2415

## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by sonichi** [comment]
This partially addresses #2423. Let's add test coverage and fix the code format error.

**Comment by cheng-tan** [comment]
existing unit test is in test_logging.py, integration test in test_agent_logging.py. It would be nice to check the schema of the json_state instead of free format

**Comment by lalo** [comment]
I went for free format of json to simplify calling site of logging, as long as it can be serialized to json I'm fine with it. In an ideal world we would have full normalized columns in this sqlite, but don't want to deal with that complexity and also don't want to restrict what types of keys and values they can log. Also, this allows same API to be used in multiple places - which is helpful for this 'event' state scenario and less for more formalized concepts like a 'new agent instance'.

File: autogen/logger/sqlite_logger.py [hunk 1]
```diff
@@ -246,6 +260,38 @@ class = excluded.class,
...
+            query = """
+            INSERT INTO events (source_id, source_name, event_name, agent_module, agent_class_name, json_state, timestamp) VALUES (?, ?, ?, ?, ?, ?, ?)
+            """
+            args = (
+                id(source),
+                source.name if hasattr(source, "name") else source,
+                name,
+                source.__module__,
+                source.__class__.__name__,
+                json.dumps(kwargs),
```
Comment (lalo) [inline autogen/logger/sqlite_logger.py:278]: todo: add catch for non serializable or default value

**Review by ekzhu — APPROVED** [review]
We can log more events in the future PRs.

**Comment by ekzhu** [comment]
@henry-zhang-bohan This PR partially address your issue regarding observability #2423 . There are separate PRs like #2329 to use different logging destination.

**Comment by sonichi** [comment]
Thanks. There is a test requiring openai CI. Could you remake a PR from the upstream repo to run that CI please? Thank you.

**Comment by lalo** [comment]
moved to https://github.com/microsoft/autogen/pull/2478

**Pull request closed without merging**